IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.m

```diff
@@ -259,7 +259,18 @@ - (BOOL)saveAccount:(nonnull MSIDAccountCacheItem *)account
     if (previousAccount)
     {
         // Make sure we copy over all the additional fields
-        [account updateFieldsFromAccount:previousAccount];
+        NSMutableDictionary *mergedDictionary = [previousAccount.jsonDictionary mutableCopy];
+        [mergedDictionary addEntriesFromDictionary:account.jsonDictionary];
+        NSError *accountError;
+        account = [[MSIDAccountCacheItem alloc] initWithJSONDictionary:mergedDictionary error:&accountError];
+        if (accountError != nil || account == nil)
```

Comment:
nit: you can just check for !account here

Reply:
updated